app/helpers/user_journey_helper.rb

```diff
@@ -30,4 +31,19 @@ def certificate_status(certificate)
   def certificate_expiry_count(msa_components, sp_components)
     (msa_components + sp_components).map(&:certificates).flatten.select(&:expires_soon?).count
   end
+
+  def sort_by_expiring_certificates(components)
```

Comment:
ok, given this doesn't need to be merged for private beta, I'd like to see some changes to where we do all of this logic... I believe it's in a wrong place outside of the models. We should approach it from the models itself, instead of the view. The view (and its helpers) should need this complexity. I propose to make the models/classes a bit more clever by:
1. adding a new attribute on to the certificate model which returns number of days left before expiry. So whenever I handle a certificate I can do `certificate.days_left` and I will get an integer.
2. adding a new attribute on the component model which returns its certificates in an expiry order. So that we can do `component.sorted_certificates`and it will return an array of certs sorted.
3. adding a new attribute on the component model which returns the number of days of the first sorted certificate from above, meaning if I do `component.days_left` I get an integer with the days left of the soonest expiring cert.
4. All of the above and in the view I don't need any logic, but will be just iterating through those new attributes... 
a. Group by environment
b. Sort by `component.days_left`
c. iterate through `component.sorted_certs`

Reply:
the additional benefit is then the each logic can be tested on each of this models as unit test and doesn't require a complicated system test/HTML wizardy...

Reply:
Thanks for all the suggestion; I am working this approach